packages/calling-component-bindings/src/captionsSelector.ts

```diff
@@ -125,21 +164,119 @@ export const captionsBannerSelector: CaptionsBannerSelector = reselect.createSel
           finalDisplayName = participant.displayName;
         }
       }
+
       return {
         id: (finalDisplayName ?? 'Unnamed Participant') + index,
         displayName: finalDisplayName ?? 'Unnamed Participant',
-        captionText: c.captionText ?? '',
-        userId
+        captionText: c.captionText,
+        userId,
+        createdTimeStamp: c.timestamp
       };
     });
+    /* @conditional-compile-remove(rtt) */
+    const completedRealTimeTexts = realTimeTexts?.completedMessages
+      ?.filter((rtt) => rtt.message !== '')
+      .map((rtt) => {
+        const userId = getRealTimeTextSpeakerIdentifier(rtt);
+        return {
+          id: rtt.id,
+          displayName: getRealTimeTextDisplayName(rtt, identifier, remoteParticipants, displayName, userId),
+          message: rtt.message,
+          userId,
+          isTyping: rtt.resultType === 'Partial',
+          isMe: rtt.isMe,
+          finalizedTimeStamp: rtt.updatedTimestamp
+        };
+      });
+    /* @conditional-compile-remove(rtt) */
+    const inProgressRealTimeTexts = realTimeTexts?.currentInProgress
+      ?.filter((rtt) => rtt.message !== '')
+      .map((rtt) => {
+        const userId = getRealTimeTextSpeakerIdentifier(rtt);
+        return {
+          id: rtt.id,
+          displayName: getRealTimeTextDisplayName(rtt, identifier, remoteParticipants, displayName, userId),
+          message: rtt.message,
+          userId,
+          isTyping: rtt.resultType === 'Partial',
+          isMe: rtt.isMe,
+          finalizedTimeStamp: rtt.updatedTimestamp
+        };
+      });
+    /* @conditional-compile-remove(rtt) */
+    const myInProgress =
+      realTimeTexts?.myInProgress && realTimeTexts.myInProgress.message !== ''
+        ? {
+            id: realTimeTexts.myInProgress.id,
+            displayName: displayName,
+            message: realTimeTexts.myInProgress.message,
+            userId: identifier,
+            isTyping: realTimeTexts.myInProgress.resultType === 'Partial',
+            isMe: true,
+            finalizedTimeStamp: realTimeTexts.myInProgress.updatedTimestamp
+          }
+        : undefined;
+
+    /* @conditional-compile-remove(rtt) */
+    // find the last final local real time text caption if myInProgress is not available
+    let latestLocalRealTimeText;
+    /* @conditional-compile-remove(rtt) */
+    if (!myInProgress) {
+      latestLocalRealTimeText =
+        realTimeTexts &&
+        realTimeTexts.completedMessages &&
+        realTimeTexts.completedMessages
+          .slice()
+          .reverse()
+          .find((rtt) => rtt.isMe);
+    }
+
     return {
-      captions: captionsInfo ?? [],
+      captions: (captionsInfo as CaptionsInformation[]) ?? [],
```

Comment:
There is a lot of type casting here, how can we avoid casting types here?

Reply:
Maybe we can explore together here to identify why these typecasts are necessary or how can we avoid them altogether.